### PR TITLE
feat(#921): D39-2.2a — backend engine + REST + auto-init SKELETON

### PR DESCRIPTION
**Pull request by zjzcpj**
Closes #921

## Summary

ADR-039 §6 Phase 1 backend SKELETON. Scaffolds the bundled-git-CLI engine, ~15 REST endpoints, the auto-init project lifecycle hooks, and the desktop portable-git fetch scripts. **All callables raise `NotImplementedError` with ≥20-line comment blocks** describing subprocess flags, parsing strategy, error envelopes, edge cases, test plan, and ADR section references so D39-2.2b can fill bodies without re-reading ADR-039.

## Owned files (NEW)

| Path | Purpose |
|---|---|
| `src/scieasy/core/versioning/__init__.py` | Public re-exports (GitEngine, GitBinary, GitChangeWatcher, write_default_gitignore, is_dirty, modified_files) |
| `src/scieasy/core/versioning/git_engine.py` | Subprocess wrapper enumerating every method needed by ADR §3.5 (init / commit / log / diff / restore / branches / branch_create / branch_switch / branch_delete / status / head_state / merge / cherry_pick / stash CRUD / merge_stage_file / merge_complete / merge_abort) |
| `src/scieasy/core/versioning/git_binary.py` | Bundle path locator + system fallback for dev CLI |
| `src/scieasy/core/versioning/gitignore_template.py` | Exact `.gitignore` template per ADR §3.3 + `write_default_gitignore()` idempotent writer |
| `src/scieasy/core/versioning/status.py` | `is_dirty()` + `modified_files()` thin wrappers |
| `src/scieasy/core/versioning/watcher.py` | `GitChangeWatcher` polling `.git/HEAD` + `refs/heads/*` for richer UI-cache invalidation (distinct from D39-2.1's engine-level emitter) |
| `src/scieasy/api/routes/git.py` | All 21 endpoint stubs from ADR §3.5 + Pydantic request/response models |
| `desktop/scripts/fetch-git-portable.ps1` | Windows MinGit fetcher (SHA-verified) |
| `desktop/scripts/fetch-git-portable.sh` | macOS universal2 + Linux musl static build (SHA-verified) |
| `tests/core/test_git_engine.py` | 17 xfail cases — engine happy paths + round-trip |
| `tests/api/test_git_endpoints.py` | 21 xfail cases — one per REST endpoint |
| `tests/cli/test_init_git_init.py` | 2 xfail cases — `scieasy init` parity + degraded mode |

## Owned files (MODIFIED skeleton stubs)

| Path | Change |
|---|---|
| `src/scieasy/api/runtime.py::create_project` | Auto-init placeholder + IMPLEMENTATION block for D39-2.2b |
| `src/scieasy/api/runtime.py::open_project` | Re-init placeholder per ADR §3.2 |
| `src/scieasy/api/runtime.py::start_workflow` | Pre-run auto-commit placeholder + cross-phase TODO for D39-2.5 (lineage join) |
| `src/scieasy/api/app.py` | Register `git_routes.router`; reserve `app.state.git_watcher = None` slot for D39-2.2b lifespan wiring |
| `src/scieasy/cli/main.py::init` | CLI git-init parity placeholder |

## Out of scope (documented)

- **`desktop/package.json`** does NOT exist yet — ADR-037 packaging pipeline pending. `fetch-git-portable.sh` carries a top-of-file note instructing the ADR-037 implementer to add `desktop/resources/git/` to the electron-builder asset list when that bundler config lands.
- **D39-2.1 already-merged work** (`workflow_watcher.py`, `ws.py`, `engine/events.py`) — UNTOUCHED.
- **Frontend** — D39-2.3a / D39-2.3b territory.
- **Bodies** — D39-2.2b territory.

## Comment-quality compliance (per dispatch prompt)

Every method in `git_engine.py` and every endpoint handler in `api/routes/git.py` carries a comment block with:
1. Purpose (paragraph)
2. Signature contract (input / output / errors)
3. Implementation steps (numbered, ≥3 steps each, specific subprocess flags + parsing rules)
4. Edge cases (≥2-3 explicit cases)
5. Test plan (bullet list of unit-test scenarios)
6. ADR references (section + line range)

## Verification

- `ruff format --check .` — clean
- `ruff check .` — clean
- `pytest tests/core/test_git_engine.py tests/api/test_git_endpoints.py tests/cli/test_init_git_init.py --timeout=60` → **43 xfailed (strict)** — every stub raises NotImplementedError as expected
- `pytest tests/api/test_app.py tests/cli/test_cli.py --timeout=60` → all 37 pass (modified files do not break existing app / CLI tests)
- All new modules import cleanly (verified with `PYTHONPATH=src python -c "import ..."`)

## Cascade checklist

D39-2.2a row ticked at `docs/planning/adr-038-039-checklist.md`. `desktop/package.json` row marked `[~]` with deferral rationale (ADR-037 pending).

### CODE REVIEW

File: src/scieasy/api/app.py [hunk 1]
```diff
@@ -205,6 +239,13 @@ def create_app() -> FastAPI:
     app.include_router(ai_pty.router)
     # ADR-036 §3.3 — server-side ruff lint endpoint for the embedded editor.
     app.include_router(lint.router)
+    # ADR-039 §3.5 — git endpoints (commit / log / diff / restore / branch
+    # ops / merge / cherry-pick / stash). Skeleton phase: all handlers
+    # raise NotImplementedError; D39-2.2b fills bodies. Router is
+    # registered now so the impl phase doesn't need to touch app.py
+    # again (and so OpenAPI surfaces the endpoints for early frontend
+    # contract work in D39-2.3a).
+    app.include_router(git_routes.router)
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/app.py]: accepted (fixed in 1054a0c): gated /api/git router behind SCIEASY_ENABLE_GIT_ROUTER env var (default OFF in skeleton phase). D39-2.2b will remove the gate when handler bodies are filled.
